### PR TITLE
Handle and report errors from TLS engine

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: library/ziti_enroll.c [hunk 1]
```diff
@@ -149,30 +149,28 @@ int ziti_enroll(ziti_enroll_opts *opts, uv_loop_t *loop, ziti_enroll_cb enroll_c
 static void well_known_certs_cb(char *base64_encoded_pkcs7, ziti_error *err, void *req) {
     ZITI_LOG(DEBUG, "base64_encoded_pkcs7 is: %s", base64_encoded_pkcs7);
 
+    int ziti_err;
     struct ziti_enroll_req *enroll_req = req;
     if ((NULL == base64_encoded_pkcs7) || (NULL != err)) {
         ZITI_LOG(DEBUG, "err->message is: %s", err->message);
         if (enroll_req->enroll_cb) {
-            enroll_req->enroll_cb(NULL, 0, err->code, enroll_req->external_enroll_ctx);
+            enroll_req->enroll_cb(NULL, ZITI_JWT_VERIFICATION_FAILED, err->code, enroll_req->external_enroll_ctx);
```
Comment (smilindave26) [inline library/ziti_enroll.c:157]: The NULL ziti_context is a bit of a downer, but fine.  Also noticed line 228 in ziti.h describes `ziti_config` parameter of `ziti_enroll_cb` as "identity json data buffer"

File: library/ziti_enroll.c [hunk 2]
```diff
@@ -181,31 +179,30 @@ static void well_known_certs_cb(char *base64_encoded_pkcs7, ziti_error *err, voi
...
+            ziti_err = ZITI_KEY_GENERATION_FAILED;
+            TRY(TLS, tls->api->generate_key(&enroll_req->ecfg->pk));
+            TRY(TLS, tls->api->write_key_to_pem(enroll_req->ecfg->pk, &enroll_req->ecfg->private_key, &len));
         }
         else {
-            tls->api->load_key(&enroll_req->ecfg->pk, enroll_req->ecfg->private_key,
-                               strlen(enroll_req->ecfg->private_key));
+            ziti_err = ZITI_KEY_LOAD_FAILED;
+            TRY(TLS, tls->api->load_key(&enroll_req->ecfg->pk, enroll_req->ecfg->private_key,
+                                        strlen(enroll_req->ecfg->private_key)));
```
Comment (smilindave26) [inline library/ziti_enroll.c:189]: strlen + 1 not needed? (the change you made to unblock me yesterday)

Comment (ekoby) [reply library/ziti_enroll.c:189]: yep, I was testing errors

Comment (ekoby) [reply library/ziti_enroll.c]: fixed

File: library/ziti_enroll.c (hunk 1 above)
Comment (ekoby) [reply library/ziti_enroll.c:157]: didn't feel right to pass object back if enrollment failed. Updated doc

**Pull request merged**